code/ImageOptimiserService.php

```diff
@@ -42,46 +42,46 @@ public function optimiseImage($filename)
         if (file_exists($filename)) {
             list($width, $height, $type, $attr) = getimagesize($filename);
 
-            $command = $this->getCommand(
-                $filename,
-                $type = $this->getImageType($type)
-            );
+            $commands = $this->getCommands($filename, $type = $this->getImageType($type));
 
-            if ($command) {
-                try {
-                    $process = $this->execCommand($command);
-                    $successful = in_array($process->getExitCode(), $this->config->get('successStatuses'));
+            if (!empty($commands)) {
```

Comment:
Since `getCommands` always returns an array, this check doesn't need to be here; if there are no commands, the foreach will do nothing anyway.